Magento_Cms/web/js/custom-country-form.js

```diff
@@ -0,0 +1,66 @@
+define(['jquery', 'mage/url', 'jquery-ui-modules/widget', 'jquery/validate'], (
+  $,
+  urlBuilder
+) => {
+  $.widget('vaimo.countryFormWidget', {
+    options: {},
+
+    _create() {
+      this.submitForm();
+    },
+
+    submitForm() {
+      const self = this;
+
+      $('#search-country-form').on('submit', function (e) {
+        e.preventDefault();
+        if ($(this).valid()) {
+          const countryId = $('#country-id').val();
+
+          self.fetchCountry(countryId);
+        }
+      });
+    },
+
+    fetchCountry(countryId) {
+      $('body').trigger('processStart');
```

Comment:
please check the last lecture where I showed how to create an AJAX request and separate everything into different methods.
this is not readable and cannot be mixined